reproducible-build-2 fails if `debuginfo-level` > 0

On my Mac OS X system, trying to run the full compiler test suite for a build with `debug = true` in the config.toml fails due to something with the run-make-fulldeps/reproducible-build-2 test.

Update: more runs without altering `debug` setting and instead doing `debuginfo-level=2` (which is implied by `debug=true`) or `debuginfo-level=1` also have the same effect. So I assume that is root cause here.

Here is the output I am currently seeing there:

```
Check compiletest suite=run-make-fulldeps mode=run-make (x86_64-apple-darwin -> x86_64-apple-darwin)

running 204 tests
ii........i...ii........i....................i................i....................i................ 100/204
....i.........i........iiii.......i...i........iiii.iii........................i...........i........ 200/204
F...
failures:

---- [run-make] run-make-fulldeps/reproducible-build-2 stdout ----

error: make failed
status: exit code: 2
command: "make"
stdout:
------------------------------------------
rm -rf /Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/test/run-make-fulldeps/reproducible-build-2/reproducible-build-2 && mkdir /Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/test/run-make-fulldeps/reproducible-build-2/reproducible-build-2
DYLD_LIBRARY_PATH="/Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/test/run-make-fulldeps/reproducible-build-2/reproducible-build-2:/Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/stage2/lib:" '/Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/stage2/bin/rustc' --out-dir /Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/test/run-make-fulldeps/reproducible-build-2/reproducible-build-2 -L /Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/test/run-make-fulldeps/reproducible-build-2/reproducible-build-2  reproducible-build-aux.rs
DYLD_LIBRARY_PATH="/Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/test/run-make-fulldeps/reproducible-build-2/reproducible-build-2:/Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/stage2/lib:" '/Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/stage2/bin/rustc' --out-dir /Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/test/run-make-fulldeps/reproducible-build-2/reproducible-build-2 -L /Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/test/run-make-fulldeps/reproducible-build-2/reproducible-build-2  reproducible-build.rs -C lto=fat
cp /Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/test/run-make-fulldeps/reproducible-build-2/reproducible-build-2/reproducible-build /Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/test/run-make-fulldeps/reproducible-build-2/reproducible-build-2/reproducible-build-a
DYLD_LIBRARY_PATH="/Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/test/run-make-fulldeps/reproducible-build-2/reproducible-build-2:/Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/stage2/lib:" '/Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/stage2/bin/rustc' --out-dir /Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/test/run-make-fulldeps/reproducible-build-2/reproducible-build-2 -L /Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/test/run-make-fulldeps/reproducible-build-2/reproducible-build-2  reproducible-build.rs -C lto=fat
cmp "/Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/test/run-make-fulldeps/reproducible-build-2/reproducible-build-2/reproducible-build-a" "/Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/test/run-make-fulldeps/reproducible-build-2/reproducible-build-2/reproducible-build" || exit 1
/Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/test/run-make-fulldeps/reproducible-build-2/reproducible-build-2/reproducible-build-a /Users/felixklock/Dev/Mozilla/rust.git/objdir-narrow/build/x86_64-apple-darwin/test/run-make-fulldeps/reproducible-build-2/reproducible-build-2/reproducible-build differ: char 334985, line 418

------------------------------------------
stderr:
------------------------------------------
make: *** [fat_lto] Error 1

------------------------------------------



failures:
    [run-make] run-make-fulldeps/reproducible-build-2

test result: FAILED. 177 passed; 1 failed; 26 ignored; 0 measured; 0 filtered out
```

It is possible that this is related to #47086

--- a/src/librustc_target/spec/apple_base.rs
+++ b/src/librustc_target/spec/apple_base.rs
@@ -31,6 +31,17 @@ pub fn opts() -> TargetOptions {
         has_elf_tls: version >= (10, 7),
         abi_return_struct_as_int: true,
         emit_debug_gdb_scripts: false,
+
+        // This environment variable is pretty magical but is intended for
+        // producing deterministic builds. This was first discovered to be used
+        // by the `ar` tool as a way to control whether or not mtime entries in
+        // the archive headers were set to zero or not. It appears that
+        // eventually the linker got updated to do the same thing and now reads
+        // this environment variable too in recent versions.
+        //
+        // For some more info see the commentary on #47086
+        link_env: vec![("ZERO_AR_DATE".to_string(), "1".to_string())],
+
         ..Default::default()
     }
 }

--- a/src/test/run-make-fulldeps/reproducible-build-2/Makefile
+++ b/src/test/run-make-fulldeps/reproducible-build-2/Makefile
@@ -2,7 +2,6 @@
 
 # ignore-musl
 # ignore-windows
-# ignore-macos (rust-lang/rust#66568)
 # Objects are reproducible but their path is not.
 
 all:  \
@@ -21,7 +20,7 @@ sysroot:
 	rm -rf $(TMPDIR) && mkdir $(TMPDIR)
 	$(RUSTC) reproducible-build-aux.rs
 	$(RUSTC) reproducible-build.rs --crate-type rlib --sysroot $(shell $(RUSTC) --print sysroot) --remap-path-prefix=$(shell $(RUSTC) --print sysroot)=/sysroot
-	cp -r $(shell $(RUSTC) --print sysroot) $(TMPDIR)/sysroot
+	cp -R $(shell $(RUSTC) --print sysroot) $(TMPDIR)/sysroot
 	cp $(TMPDIR)/libreproducible_build.rlib $(TMPDIR)/libfoo.rlib
 	$(RUSTC) reproducible-build.rs --crate-type rlib --sysroot $(TMPDIR)/sysroot --remap-path-prefix=$(TMPDIR)/sysroot=/sysroot
 	cmp "$(TMPDIR)/libreproducible_build.rlib" "$(TMPDIR)/libfoo.rlib" || exit 1

